server/src/main/java/org/elasticsearch/rest/action/cat/RestIndicesAction.java

```diff
@@ -422,9 +425,8 @@ Table buildTable(RestRequest request, IndexMetaData[] indicesMetaData, ClusterHe
             final CommonStats primaryStats;
             final CommonStats totalStats;
 
-            if (state == IndexMetaData.State.CLOSE) {
+            if (indexState == IndexMetaData.State.CLOSE) {
                 // empty stats for closed indices, but their names are displayed
-                assert indexStats == null;
```

Comment:
do closed indices now give an indexStats response?

Reply:
Only in the context of the Cat Indices API which works by executing a Cluster State request, then a Cluster Health request and finally an Indices Stats request. All requests now share the same indices options (`strictExpand`) which expand to opened and closed indices and does not forbid closed indices.

The Indices Stats API uses the `strictExpandOpenAndForbidClosed` indices option by default and it will forbids closed indices to be included in stats, unless the Java API is used to pass a different indices options that allow them - like the Cat Indices API does.

Note that stats for closed indices are almost always `0`, except for docs stats that are initialized in the read only engine ctor.